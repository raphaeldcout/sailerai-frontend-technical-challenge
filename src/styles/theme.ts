const theme = {
    colors: {
        primary: '#183145',
        primaryLight: '#788AA3',
        primaryDark: '#212121',
        primaryMedium: '#11283a',
        secondary: '#38b3ff',
        background: '#1f1f22',
        surface: '#f5f5f5',
        error: '#d32f2f',
        success: '#388e3c',
        successLight: '#2DD881',
        warning: '#fbc02d',
        border: '#e0e0e0',
        light: '#ffffff',
        neutralLowLight: '#b0d4ee',
        neutralLowMedium: '#f1f1f1',
        neutralLowDark: '#c4c4c4',
        neutralMediumLight: '#6c757d',
        neutralMedium: '#495057',
        greenLowLight: '#dcf6c6',
        rbg: {
            greenLowMedium: 'rgba(46, 178, 110, 0.2)',
            greenLowDark: 'rgba(46, 178, 110, 0.3)',
        }
    },
    darkMode: {
        foregroundRgb: '255, 255, 255',
        backgroundStartRgb: '0, 0, 0',
        backgroundEndRgb: '19, 19, 22',
    },
    spacings: {
        xs: '4px',
        sm: '8px',
        md: '16px',
        lg: '24px',
        xl: '32px',
        xxl: '64px',
    },
    typography: {
        fontSize: {
            xs: '12px',
            sm: '14px',
            md: '16px',
            lg: '20px',
            xl: '24px',
            xxl: '32px',
        },
        fontWeight: {
            light: 300,
            normal: 400,
            medium: 500,
            bold: 700,
        },
        lineHeight: {
            xs: '1rem',
            sm: '1.2rem',
            md: '1.4rem',
            lg: '1.6rem',
            xl: '1.8rem',
        },
    },
    borders: {
        radius: {
            none: '0px',
            sm: '4px',
            md: '8px',
            lg: '12px',
            full: '999px',
        },
        width: {
            thin: '1px',
            thick: '2px',
        },
    },
    shadows: {
        sm: '0px 1px 3px rgba(0, 0, 0, 0.12)',
        md: '0px 3px 6px rgba(0, 0, 0, 0.16)',
        lg: '0px 10px 20px rgba(0, 0, 0, 0.2)'
    },
    zIndex: {
        dropdown: 1000,
        modal: 1100,
        tooltip: 1200,
    },
    breakpoints: {
        xsmall: 575.98,
        small: 767.98,
        medium: 991.98,
        large: 1199.98,
        xlarge: 1399.98
    }
};

export const colors = theme.colors;

export type Theme = typeof theme;

export type ColorsType = keyof Theme['colors'];

export default theme;