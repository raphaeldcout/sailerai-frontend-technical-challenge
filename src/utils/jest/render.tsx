import { FC, ReactElement, ReactNode } from 'react';

import { ThemeProviderSailer } from '@/styles/withThemeProvider';
import { render, RenderOptions } from '@testing-library/react';

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <ThemeProviderSailer>{children}</ThemeProviderSailer>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };

