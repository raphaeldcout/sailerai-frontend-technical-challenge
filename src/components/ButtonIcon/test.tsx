import '@testing-library/jest-dom';

import { FiSend } from 'react-icons/fi';

import theme from '@/styles/theme';
import { render } from '@/utils/jest/render';

import { ButtonIcon } from '.';

describe('ButtonIcon', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <ButtonIcon>
        <FiSend size={20} color={theme.colors.error} />
      </ButtonIcon>
    );

    expect(container).toMatchSnapshot();
  });
});
