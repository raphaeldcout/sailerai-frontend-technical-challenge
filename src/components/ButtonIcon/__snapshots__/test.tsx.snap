// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon should match snapshot 1`] = `
<div>
  <button
    backgrouncolor="primaryLight"
    class="sc-95866428-0 kXqgKZ"
  >
    <svg
      color="#d32f2f"
      fill="none"
      height="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      style="color: rgb(211, 47, 47);"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="22"
        x2="11"
        y1="2"
        y2="13"
      />
      <polygon
        points="22 2 15 22 11 13 2 9 22 2"
      />
    </svg>
  </button>
</div>
`;
